static/js/storybook/index.js

```diff
@@ -135,3 +191,153 @@ storiesOf("Links and Buttons", module)
       </StoryWrapper>
     )
   })
+
+const fakeUrl = (prefix, id) =>
+  `https://avatars.dicebear.com/v2/identicon/${prefix}_${id}.svg`
+
+storiesOf("Post", module)
+  .addDecorator(withSetup)
+  .addDecorator(withKnobs)
+  .add("compact post", () => {
+    const isUrl = boolean("Is URL post", false)
+    const post = makePost(isUrl)
+    post.profile_url = fakeUrl("pr", post.author_id)
+    if (isUrl) {
+      post.thumbnail = fakeUrl("th", post.author_id)
+    }
+
+    const isModerator = boolean("Is moderator", false)
+    const isAnonymous = boolean("Is anonymous", false)
+    if (!isAnonymous) {
+      SETTINGS.username = "user"
+    }
+
+    const showPinUI = boolean("Show pin UI", false)
+    post.stickied = showPinUI
+
+    return (
+      <StoryWrapper>
+        <CompactPostDisplay
+          post={post}
+          menuOpen={boolean("Menu open", false)}
+          showChannelLink={boolean("Show channel link", false)}
+          showPinUI={showPinUI}
+          removePost={action("remove post")}
+          ignorePostReports={action("ignore post reports")}
+          reportPost={action("report post")}
+          togglePinPost={action("pin")}
+          isModerator={isModerator}
+        />
+      </StoryWrapper>
+    )
+  })
```

Comment:
I might split this out into a few stories, like `url post`, `compact post`, `pinned post`, etc. that way storybook can start to function as documentation of the different UI states we support for the component.

to make that easier it might make sense to write a function which takes a few parameters and returns a story. I didn't do that for the simpler stories I wrote (for the link and so on) because I didn't think it was worth it, but for something more complicated like this it might make sense to do so.

Reply:
Do you have an idea what should be a knob and what should be split into separate stories?

Reply:
I think if we have components where we have a few different designed UI states then those should be different stories, like link posts vs text posts, etc. It's a bit of a judgment call, but that's where I'd say the line is :woman_shrugging: